src/app_service/service/contacts/service.nim

```diff
@@ -296,15 +296,35 @@ QtObject:
       else:
         contact.blocked = false
 
-      let response = status_contacts.addContact(contact.id, contact.name)
+      let response = status_contacts.sendContactRequest(contact.id, message)
+     # let response = status_contacts.addContact(contact.id, contact.name, message)
+      if(not response.error.isNil):
+        let msg = response.error.message
+        error "error sending contact request", msg
+        return
+      self.saveContact(contact)
+      self.events.emit(SIGNAL_CONTACT_ADDED, ContactArgs(contactId: contact.id))
+    except Exception as e:
+      error "an error occurred while sending contact request", msg=e.msg
+
+  proc acceptContactRequest*(self: Service, publicKey: string) =
+    try:
+      # TODO: does we need this logic on both (sendContactRequest and acceptContactRequest) sides?
```

Comment:
@saledjenic should i save (and refactor to other method) this logic for both cases?

Reply:
@MishkaRogachev as we discussed in private fill free to update logic here according to the new changes introduced on the status-go side.